xdsl/pattern_rewriter.py

```diff
@@ -372,22 +397,22 @@ def op_type_rewrite_pattern(
     if not is_method:
 
         def op_type_rewrite_pattern_static_wrapper(
-                op: Operation, rewriter: PatternRewriter) -> None:
+                self: RewritePattern, op: Operation,
+                rewriter: PatternRewriter) -> None:
             if not isinstance(op, expected_type):
                 return None
-            func(op, rewriter)
+            func(op, rewriter)  # type: ignore
 
         return op_type_rewrite_pattern_static_wrapper
 
     def op_type_rewrite_pattern_method_wrapper(
-            self,  # type: ignore
-            op: Operation,
+            self: _RewritePatternT, op: Operation,
             rewriter: PatternRewriter) -> None:
         if not isinstance(op, expected_type):
             return None
-        func(self, op, rewriter)
+        func(self, op, rewriter)  # type: ignore
```

Comment:
Instead of this "type: ignore", can we cast the type of `func` to the expected type (three args)?